flaps/flaps.go

```diff
@@ -549,15 +555,46 @@ func (f *Client) UnCordon(ctx context.Context, machineID string) (err error) {
 	return nil
 }
 
+type flapsCall struct {
+	Call     string  `json:"c"`
+	Command  string  `json:"co"`
+	Duration float64 `json:"d"`
+}
+
+var re = regexp.MustCompile(`\/(?P<machineId>[a-z-A-Z0-9]*)\/(?P<flapsCall>.*)`)
+
 func (f *Client) sendRequest(ctx context.Context, method, endpoint string, in, out interface{}, headers map[string][]string) error {
 	timing := instrument.Flaps.Begin()
+	invocationID := metrics.InvocationIDFromContext(ctx)
+
+	defer func() {
+		matches := re.FindStringSubmatch(endpoint)
+		index := re.SubexpIndex("flapsCall")
+
+		// unless something changes about flaps, this should always be false
+		if index >= len(matches) {
+			return
+		} else {
+			err := fmt.Errorf("flaps endpoint %s did not match regex %s", endpoint, re.String())
+			sentry.CaptureException(err)
+		}
+
+		call := matches[index]
+		metrics.Send(ctx, "flaps_call", flapsCall{
+			Call:     call,
+			Command:  command_context.FromContext(ctx).Name(),
+			Duration: time.Since(timing.Start).Seconds(),
+		})
+
+	}()
 	defer timing.End()
 
 	req, err := f.NewRequest(ctx, method, endpoint, in, headers)
 	if err != nil {
 		return err
 	}
 	req.Header.Set("User-Agent", f.userAgent)
+	req.Header.Set("Invocation-ID", invocationID)
```

Comment:
The thing is that we generate these IDs per invocation of flyctl, as opposed to doing them per flaps request specifically. This will hopefully let us correlate different requests between 1 user's "action".